### PR TITLE
streamtest: add option to add a correct base address

### PR DESCRIPTION
**Pull request by acud**
This rectifies the mistake we had in the recorder, where the `p2p.Peer` that we called the handler call with was the wrong address, carrying over the destination address, instead of the origin address of the message.

There is still one todo here, and that is to rewrite the preemptive retry test to use mocked server responses instead of the current behavior which is injected through a middleware if statement. Since this change is a blocker for @esadakar's PR, I'd rather have it in so he can rebase, then we can mend the skipped test.

### CODE REVIEW

File: pkg/accounting/mock/accounting.go [hunk 1]
```diff
@@ -153,6 +154,7 @@ func (s *Service) Debit(peer swarm.Address, price uint64) error {
 
 // Balance is the mock function wrapper that calls the set implementation
 func (s *Service) Balance(peer swarm.Address) (*big.Int, error) {
+	fmt.Println(s.balances)
```
Comment (esadakar) [inline pkg/accounting/mock/accounting.go:157]: I can the print statement in my PR

**Review by esadakar — APPROVED** [review]

File: pkg/accounting/mock/accounting.go (hunk 1 above)
Comment (acud) [reply pkg/accounting/mock/accounting.go]: reverted :+1:

File: pkg/pushsync/pushsync_test.go [hunk 2]
```diff
@@ -358,8 +357,9 @@ func TestHandler(t *testing.T) {
 		t.Fatalf("unexpected balance on trigger. want %d got %d", -int64(fixedPrice), balance)
 	}
 
-	// we need to check here for pivotPeer instead of triggerPeer because during streamtest the peer in the handler is actually the receiver
-	balance, err = pivotAccounting.Balance(pivotPeer)
+	// we need to check here for pivotPeer instead of triggerPeer because during
```
Comment (ralph-pichler) [inline pkg/pushsync/pushsync_test.go:360]: you may want to remove the comment as the hack is no longer necessary.

**Review by janos — APPROVED** [review]
LGTM

**Pull request merged**